Automattic/wp-calypso
# WIP: Dependency finder for smarter builds

#### Changes proposed in this Pull Request
This is the initial implementation of the dependency finder package. The ultimate goal is to be able to take a certain CI/CD build x, and then determine when it should run. Builds should run when files which would change the build output are modified.

This package attempts to find all files which could possibly affect the build output of a given package. Here are some items it finds:
- All local project imports (e.g. "found files")
- All imported npm packages (including monorepo ones)

It also lists files which don't fit into the above definitions:
- Files which exist in the filesystem, but do not appear to be imported. (Like a bash script or configuration file.)
- Files which were imported, but could not be located on the filesystem.


### To Do
- [ ] Solve all reported "missing files" 
    - [ ] Dependencies imported via dependency extraction plugin (a8c-fse-common-data-stores)
    - [ ] Some bug with the whats-new imports in ETK

#### Testing instructions
Run this in your shell to test just for ETK. Leave out the apps/editing-toolkit portion if you want to see for all packages/apps.

```
yarn workspace @automattic/dependency-finder build
node ./packages/dependency-finder/dist/esm/index.js apps/editing-toolkit
```

You should see all the files it detects in the different categories.
